### DISCUSSION TITLE
How do I redirect from getServerSideProps with a callback url?

### DISCUSSION THREAD

**Discussion by gopinav** [nextauthjs/next-auth#2134] [Help]
I'm trying to add a 'route guard' if you can call it that by checking if session is not null in getServerSideProps and redirecting to the built in signin page. Although that works, I'm always redirected to the home page after signing in. I tried to add the callbackUrl query parameter as you can see below but it still doesn't work as expected. How do I achieve this? Thanks in advance.

blog.js
```
export async function getServerSideProps(context) {
  const session = await getSession(context)
  if (!session) {
    return {
      redirect: {
        destination: `/api/auth/signin?callbackUrl=${encodeURIComponent(
          'http:localhost:3000/blog'
        )}`,
        permanent: false
      }
    }
  }
  return {
    props: {
      data: 'Blog data',
      session
    }
  }
}
```

**Comment by srijans38** [comment]
Hey @gopinav, It seems like you have a typo. Your callbackUrl parameter should be `http://localhost:3000/blog` but its currently `http:localhost:3000/blog` i.e. missing slashes after `http:`. Correcting this should fix the issue.

Callback url is being checked against the base url of the app using String.prototype.startsWith() as far as I know and that's what's causing this behaviour as it defaults to just the base url. Hence, you are redirected to the home page.
